sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -620,8 +620,17 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       buildCast[UTF8String](_, s => if (s.toInt(result)) result.value else null)
     case BooleanType =>
       buildCast[Boolean](_, b => if (b) 1 else 0)
+    case DateType if ansiEnabled =>
```

Comment:
I might be missing something but will this branch ever execute? the case DateType above would seem to cover this case too

Reply:
> I might be missing something but will this branch ever execute? the case DateType above would seem to cover this case too

All handle like timestamp type, cc @cloud-fan

Reply:
This cast s not allowed in ANSI mode, why do we need this code path?